pennylane/math/__init__.py

```diff
@@ -76,6 +76,25 @@
 toarray = ar.numpy.to_numpy
 T = ar.numpy.transpose
 
+
+def array(*args, like=None, **kwargs):
+    """The autoray.numpy.array method with the extension that the Torch devices
+    are preserved."""
+    res = ar.numpy.array(*args, like=like, **kwargs)
+    if like is not None and get_interface(like) == "torch":
+        res = res.to(device=like.device)
+    return res
+
+
+def eye(*args, like=None, **kwargs):
+    """The autoray.numpy.eye method with the extension that the Torch devices
+    are preserved."""
+    res = ar.numpy.eye(*args, like=like, **kwargs)
+    if like is not None and get_interface(like) == "torch":
+        res = res.to(device=like.device)
+    return res
```

Comment:
I think here it would make more sense to use the `single_dispatch.py` file to define those functions.

Reply:
Moved these to `multi_dispatch.py`, that contained similar functions.